src/metamask-context.ts

```diff
@@ -57,16 +57,36 @@ export type IMetaMaskContext = MetaMaskState & {
    */
   connect: () => Promise<string[] | null>;
   /**
-   * Request addition of a new network
-   * @param parameters New chain parameters, see [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for full description
+   * Request addition of a new chain to MetaMask
+   * @dev See [MetaMask API](https://docs.metamask.io/wallet/reference/rpc-api/#wallet_addethereumchain) or [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for full description
+   * @param parameters The chain parameters
+   * @example ```ts
+   * const { addChain } = useMetaMask();
+   * const GNOSIS_MAINNET_PARAMS = {
+   *  chainId: "0x64",
+   *  chainName: "Gnosis",
+   *  nativeCurrency: {
+   *    name: "xDai",
+   *    symbol: "XDAI",
+   *    decimals: 18,
+   *  },
+   *  rpcUrls: ["https://rpc.gnosischain.com/"],
+   *  blockExplorerUrls: ["https://gnosisscan.io/"],
+   * }
+   * // Add Gnosis chain to MetaMask
+   * const onClick = () => addChain(GNOSIS_MAINNET_PARAMS);
+   * ```
    */
   addChain: (parameters: AddEthereumChainParameter) => Promise<void>;
   /**
    * Request a switch of network
+   * @dev See [MetaMask API](https://docs.metamask.io/wallet/reference/rpc-api/#wallet_switchethereumchain) or [EIP-3326](https://ethereum-magicians.org/t/eip-3326-wallet-switchethereumchain/5471) for full description
+   * @dev An error with code `4902` will be thrown if the chain has not been added to MetaMask, in this case, use `addChain` first
```

Comment:
There is no way check if the network has been added and that function would success or not. So "use `addChain` first" is a bit confusing.

Reply:
I'm thinking there could be a third function `switchOrAddChain(chainId, parameters: AddEthereumChainParameter)` that handles the chain addition and further switching

Reply:
Replied in the issue. I agree that documentation is good improvement for now